### DISCUSSION TITLE
CORS Error When Attempting Cross-Origin Authentication with FastAPI Integration

### DISCUSSION THREAD

**Discussion by Pediatra** [fief-dev/fief#487] [Q&A] [answered]
I'm following the[ Web Application tutorial for FastAPI integration](https://docs.fief.dev/integrate/python/fastapi/#web-application-example), but instead of using FastAPI to serve HTML, I'm trying to set a cookie for API authentication purposes.

However, when attempting to make a fetch request from the frontend to the API for authentication the endpoint tries to redirect to the authentication url but the following CORS error appears:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:8000/authorize?.... (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 400.
```

To resolve this, I’ve tried modifying the ALLOW_ORIGIN_REGEX environment setting to allow all origins, but the issue persists, and I continue receiving the CORS error.

Could you provide guidance on how to properly handle cross-origin requests for authentication in this setup, or if there is a way to enable cross-origin redirection in this case?

**Comment by fief-bailiff** [comment]
Hail, @Pediatra 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment, marked as answer]
The authorize page is not designed to be fetched by JavaScript: you should redirect the user to this page.

Actually, even if you don't serve HTML through FastAPI, the process is the same: the backend will set the cookie and redirect the user to a page. You can very well redirect to your frontend app at this stage.

Another approach is to let the frontend app get the token and pass it manually to the backend, as explained in the browser example: https://docs.fief.dev/integrate/javascript/frontend/browser/
